app/models/account/trade_builder.rb

```diff
@@ -31,6 +31,7 @@ def create_entry
     end
 
     def security
+      return Security.find(ticker) if ticker.match?(/\A[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}\z/i)
       Security.find_or_create_by(ticker: ticker)
```

Comment:
@Shpigford @zachgoll you might want to have some additional processing here.

Because you're returning search results on company names (in addition to ticker symbols), and also allowing free-text entry, people might enter e.g. "Apple". They'll see that the combobox returned an equivalent result, and blur the combobox without selecting anything.

At that point the combobox will think they meant to enter the literal string "Apple", and not "AAPL". I think that'll result in a Security with ticker "Apple" being created globally. But I haven't taken the time to really understand this builder class yet.

Might not be a problem in practice, just wanted to point that out.

Reply:
This is a tricky one because for self-hosters who won't have the securities loaded yet, we'd still need to support the free-text entry. 

That said, I think @Shpigford is working on a slightly different data model for securities + prices right now that may make it so we can be more strict with this input—i.e. only allowing IDs to be passed to the input's value and throwing a `required` client-side validation if not selected.

Reply:
That new data model solution sounds good!

If and when you switch to that, changing the combobox to `autocomplete: :both` (the default) instead of `:list` will always select the best match, too. So it'll help with the `required` UX.

Reply:
@josefarias awesome, will do.  Thanks for the heads up!